### PR TITLE
Dropping support for symfony < 3.4

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

This is the same thing as https://github.com/sonata-project/SonataAdminBundle/pull/5733, dropping support of Symfony < 3.4 and >= 4, < 4.2

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Removed
- Support for Symfony < 3.4
- Support for Symfony >= 4, < 4.2
```

There are some changes that weren't marked as deprecated:

- [Removing `getName` from Form](https://github.com/symfony/symfony/blob/3.4/UPGRADE-3.0.md#form)

### CODE REVIEW

File: tests/Controller/CategoryAdminControllerTest.php [hunk 1]
```diff
@@ -189,50 +178,26 @@ protected function setUp()
...
-                    return 'csrf-token-123_'.$intention;
-                });
-
-            $this->csrfProvider->expects($this->any())
-                ->method('isCsrfTokenValid')
-                ->willReturnCallback(static function ($intention, $token) {
-                    if ($token === 'csrf-token-123_'.$intention) {
-                        return true;
-                    }
+        $this->csrfProvider = $this->getMockBuilder(
```
Comment (OskarStark) [inline tests/Controller/CategoryAdminControllerTest.php]: Can we use createMock() here?

File: src/Form/Type/CategorySelectorType.php [hunk 2]
```diff
@@ -42,32 +39,10 @@ public function __construct(ManagerInterface $manager)
         $this->manager = $manager;
     }
 
-    /**
-     * NEXT_MAJOR: Remove method, when bumping requirements to SF 2.7+.
-     */
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
```
Comment (phansys) [inline src/Form/Type/CategorySelectorType.php]: Removing a public method breaks BC.

Comment (phansys) [reply src/Form/Type/CategorySelectorType.php]: We should deprecate these methods.

**Comment by greg0ire** [comment]
Thanks @franmomu !

**Pull request merged**